Cognite.Simulator.Utils/FileStorageClient.cs

```diff
@@ -29,38 +33,65 @@ public FileStorageClient(HttpClient client, ILogger<FileStorageClient> logger)
         /// Downloads the file from the provided <paramref name="uri"/> and saves
         /// it in the provided <paramref name="filePath"/>
         /// </summary>
-        /// <param name="uri"></param>
-        /// <param name="filePath"></param>
+        /// <param name="uri">URI to download from</param>
+        /// <param name="filePath">Path to save the file</param>
+        /// <param name="failLargeFiles">Set to true to throw an exception if the file is larger than <see cref="LargeFileSize"/></param>
         /// <returns><c>true</c> if success, else <c>false</c></returns>
-        public async Task<bool> DownloadFileAsync(Uri uri, string filePath)
+        public async Task<bool> DownloadFileAsync(Uri uri, string filePath, bool failLargeFiles = false)
         {
             try
             {
-                var response = await _client.GetAsync(uri)
+                var response = await _client.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead)
                     .ConfigureAwait(false);
                 if (!response.IsSuccessStatusCode)
                 {
                     _logger.LogWarning("Failed to download the file: {Message}", response.ReasonPhrase);
                     return false;
                 }
 
+                if (response.Content.Headers.ContentLength > MaxFileDownloadSize)
+                {
+                    _logger.LogWarning("File size exceeds the maximum allowed size: {Size} bytes", MaxFileDownloadSize);
+                    return false;
+                }
+
+                if (failLargeFiles && response.Content.Headers.ContentLength > LargeFileSize)
+                {
+                    throw new ConnectorException(
+                        $"The file is larger than {LargeFileSize / 1024 / 1024} MB and needs to be pre-downloaded before any operations can be performed." +
+                        "Connector hasn't been able to pre-download the file yet. " +
+                        "Please wait for the file to be downloaded and try again."
+                    );
+                }
+
                 if (File.Exists(filePath))
                 {
                     File.Delete(filePath);
                 }
-                using (var fs = new FileStream(filePath, FileMode.CreateNew))
+
+                using (var responseStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false))
+                using (var fileStream = new FileStream(filePath, FileMode.CreateNew, FileAccess.Write, FileShare.None, bufferSize: DefaultBufferSize, useAsync: true))
```

Comment:
why did we add these new parameters? `FileAccess.Write, FileShare.None,`

Is it because of `useAsync`?

Reply:
because of the `bufferSize` param